components/raftstore/src/store/async_io/write.rs

```diff
@@ -189,16 +262,55 @@ where
     }
 }
 
+/// These states are set only in raftstore V2.
+#[derive(Default)]
+pub struct ExtraStates {
```

Comment:
why not extend it and let v1 reuse it

Reply:
Actually there is a TODO https://github.com/tikv/tikv/pull/13157/files#diff-2b07a6df127983bc0dac285d2ca79b6762ae50749697441f11420fafe7ebd21bR89. But it's not a goal of the PR and I don't want to add more complexity to V1.

Reply:
Yes, but the complexity is restricted inside writer and and the old code path will still be the same. And after wards, it's free to mutate the v2 implementations and add more asynchronous implementations like async destroy.